### PR TITLE
Support LDAP search filter

### PR DESCRIPTION
**Pull request by thasegaw**
Hi, I'd like to use LDAP filter option.
This PR add the `ldap-search-filter` option for `auth-ldap`.  

The documentation of `parseFilter` is http://ldapjs.org/filters.html

### CODE REVIEW

File: lib/util/ldaputil.js [hunk 1]
```diff
@@ -60,6 +60,16 @@ module.exports.login = function(options, username, password) {
           })
         }
 
+    if (options.search.filter !== 'undefined') {
```
Comment (sorccu) [inline lib/util/ldaputil.js]: Did you mean to use 'typeof' here?

File: lib/util/ldaputil.js [hunk 2]
```diff
@@ -44,20 +44,16 @@ module.exports.login = function(options, username, password) {
 
   function tryFind(client) {
     var resolver = Promise.defer()
+    var filter = `&(objectClass=${options.search.objectClass})` +
+                 `(${options.search.field}=${username})`
```
Comment (sorccu) [inline lib/util/ldaputil.js]: Using string filters with user-defined content is vulnerable to [LDAP Injection](https://www.owasp.org/index.php/LDAP_injection). Please use the filter builders instead.

**Comment by thasegaw** [comment]
@sorccu Thank you for your comment!
I have fixed some bugs and tested both simple and complex filter.

**Comment by denis99999** [comment]
Great, when this feature will be available in the master branch ?

**Comment by sorccu** [comment]
Oops, looks like I forgot again. Will merge later tonight.

**Comment by denis99999** [comment]
Thanks !

**Pull request merged**